### PR TITLE
Tailwind config: also look for .tsx .jsx and .ts

### LINKED ISSUE #14595: Tailwind CSS base utilities not applied to production builds for TypeScript projects
# Bug report

## Describe the bug

Production builds of Next.js do not apply the base Tailwind utilities in the static build if any components in the `pages/` directory imports any external TypeScript components (such as `components/nav.tsx`), the build only apply the styling added into the

## To Reproduce

1. Create a new NextJS project using `yarn create next-app project-name`
1. Select `with-tailwindcss` example project
1. Add required TypeScript dependencies `yarn add --dev typescript @types/react @types/node`
1. Rename `components/nav.js` to `components/nav.tsx`
1. Run `yarn run build && yarn run start`
1. Navigate to `http://localhost:3000`

## Expected behavior

That it would leverage the base Tailwind CSS utilities and appear as the default rather than the screen shot below.

## Screenshots

The base utilities, such as flexbox, are lost. Only the styles using the `@apply` attribute in `styles/index.css` are applied.
Screenshot_2020-06-26 Screenshot

## System information

- OS: macOS
- Version of Next.js: 9.2.0
- Version of Node.js: 14.4.0

## Additional context

This doesn't seem to happen if any page components are using TypeScript, only if the components that the page components import are using TypeScript.

### PR DESCRIPTION
**Pull request by IamLizu**
Closes: #14595

### CODE REVIEW

File: examples/with-tailwindcss/tailwind.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  purge: ['./components/**/*.js', './pages/**/*.js'],
+  purge: ['./components/**/*.js', './pages/**/*.js', './components/**/*.jsx', './pages/**/*.jsx', './components/**/*.ts', './pages/**/*.ts', './components/**/*.tsx', './pages/**/*.tsx'],
```
Comment (timneutkens) [inline examples/with-tailwindcss/tailwind.config.js:2]: I think you could potentially do this, haven't tried it though:

```suggestion
  purge: ['./components/**/*.{js,ts,jsx,tsx}', './pages/**/*.{js,ts,jsx,tsx}'],
```

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 11.5s | 11.5s | ⚠️ +50ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.003 | 2.042 | ⚠️ +0.04 |
| / avg req/sec | 1248.25 | 1224.39 | -23.86 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.313 | 1.447 | ⚠️ +0.13 |
| /error-in-render avg req/sec | 1904.3 | 1727.72 | -176.58 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..558d.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.1 kB | 57.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.59 kB | 5.59 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.08 kB | 7.08 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.6 kB | 52.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 952 B | 952 B | ✓ |
| link.html gzip | 957 B | 957 B | ✓ |
| withRouter.html gzip | 944 B | 944 B | ✓ |
| Overall change | 2.85 kB | 2.85 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12.2s | -117ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..558d.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.1 kB | 57.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.59 kB | 5.59 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.08 kB | 7.08 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.6 kB | 52.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 876 kB | 876 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 908 kB | 908 kB | ✓ |
| withRouter.js | 908 kB | 908 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>
Commit: e6194f52aebe23d410408ecda6ffb02c7b4921f3

**Comment by timneutkens** [comment]
@IamLizu did you verify the suggested change works?

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 11.1s | 11.4s | ⚠️ +280ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.994 | 1.989 | 0 |
| / avg req/sec | 1253.71 | 1256.98 | ⚠️ +3.27 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.294 | 1.308 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 1932.24 | 1911.94 | -20.3 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..558d.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.1 kB | 57.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.59 kB | 5.59 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.08 kB | 7.08 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.6 kB | 52.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 952 B | 952 B | ✓ |
| link.html gzip | 957 B | 957 B | ✓ |
| withRouter.html gzip | 944 B | 944 B | ✓ |
| Overall change | 2.85 kB | 2.85 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 11.7s | 11.7s | ⚠️ +3ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..558d.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.1 kB | 57.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.59 kB | 5.59 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.08 kB | 7.08 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.6 kB | 52.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | IamLizu/next.js patch-1 | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 876 kB | 876 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 908 kB | 908 kB | ✓ |
| withRouter.js | 908 kB | 908 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>
Commit: 66ead70de12f4f6e4e51eeb8376ee9f06132b70c

**Comment by IamLizu** [comment]
> @IamLizu did you verify the suggested change works?

Yes, I did, here's the fixed result.
image

**Review by timneutkens — APPROVED** [review]
Thanks for checking!

**Pull request merged**